charts/ingress-nginx/values.yaml

```diff
@@ -76,15 +76,13 @@ controller:
   ##
   electionID: ingress-controller-leader
 
-  ## Name of the ingress class to route through this controller
-  ##
-  ingressClass: nginx
-
   # This section refers to the creation of the IngressClass resource
-  # IngressClass resources are supported since k8s >= 1.18
+  # IngressClass resources are supported since k8s >= 1.18 and required since k8s >= 1.19
   ingressClassResource:
-    enabled: false
+    name: nginx
+    enabled: true
     default: false
```

Comment:
because it's required since k8s >= 1.19, should we change the default value to `true`?

Reply:
Will answer just this and let the others to later:

The idea of default here being false is because when you set it to true, it will assume every new Ingress resource created in the cluster must have / use that ingress class. This is bad, if you have other controllers running in the cluster.

Imagine the same situation with StorageClass, where you have a default storageclass, and then a Helm chart adds another storageclass saying "hey, this is the new default".

Now, two things may happen: Cluster will refuse to create new ingress resources if you have two default classes (I guess this is the default behavior) or, if you don't have any default class, but a bunch of other controllers, this helm chart will take over.

IMO this should be an opt-in, and we need to document really well in the migration guide. For the majority of users, it's fine and also great to have a default ingressclass if they run only one ingress controller :)

I'm worried about the corner cases

Reply:
I agree with you, then we need to add this description in the document.